src/shader.cpp

```diff
@@ -10,6 +10,112 @@
 
 namespace bgfx
 {
+	struct DescriptorTypeToId
+	{
+		DescriptorType::Enum type;
+		uint16_t id;
+	};
+
+	static DescriptorTypeToId s_descriptorTypeToId[] =
+	{
+		// NOTICE:
+		// DescriptorType must be in order how it appears in DescriptorType::Enum! id is
+		// unique and should not be changed if new DescriptorTypes are added.
+		{ DescriptorType::Undefined,     0xFFFF },
```

Comment:
Please remove VK specific comments from backend independent code.

Also why not start from 0, 1, 2, etc?

Reply:
The values are taken from the VK/WebGPU enums that have been used directly so far so that existing shaders don't have to be recompiled or versioning magic has to be introduced. Comments indicate the original enum values, but I suppose that's not necessary to know.

Reply:
Ah ok, so you're basically making it compatible with previous binary.